mergify_engine/tasks/engine/__init__.py

```diff
@@ -65,24 +66,20 @@ def get_github_pull_from_event(repo, event_type, data):
             pulls = data["check_suite"]["pull_requests"]
             sha = data["check_suite"]["head_sha"]
         if not pulls:
-            return get_github_pull_from_sha(repo, sha)
+            pulls = [get_github_pull_from_sha(client, sha)]
         if len(pulls) > 1:  # pragma: no cover
             # NOTE(sileht): It's that technically possible, but really ?
             LOG.warning(
                 "check_suite/check_run attached on multiple pulls",
-                gh_owner=repo.owner.login,
-                gh_repo=repo.name,
+                gh_owner=client.owner,
+                gh_repo=client.repo,
                 event_type=event_type,
             )
 
-        for p in pulls:
-            try:
-                pull = repo.get_pull(p["number"])
-            except github.UnknownObjectException:  # pragma: no cover
-                continue
-
-            if pull and not pull.merged:
-                return pull
+        # NOTE(sileht): data here is really incomplete most of the time,
+        # but mergify will complete it automatically if mergeable_state is missing
+        for pull in pulls:
+            return pull
```

Comment:
This looks suspicious… just `return pulls[0]`?

Reply:
In aA previous change, `pulls` was a iterator, but this is not needed anymore